taskfile/task.go

```diff
@@ -90,6 +119,7 @@ func (t *Task) UnmarshalYAML(node *yaml.Node) error {
 			Prefix        string
 			IgnoreError   bool `yaml:"ignore_error"`
 			Run           string
+			On            []string
```

Comment:
You could implement this as `[]Platform` and then use the [`Unmarshaler` interface](https://pkg.go.dev/gopkg.in/yaml.v2#Unmarshaler) to process it rather than calling `processPlatforms` manually. This might be a bit cleaner.

I would personally also move the `Platform` struct and its `UnmarshalYAML` function to their own `.go` file in the `taskfile` package. Similar to what has been done for `precondition` or `vars`.

Reply:
Good suggestions. I was on the verge of making another package anyway so that's fine. I haven't done a yaml unmarshaller before but I'm sure it's easy enough...